src/precice/bindings/python/precice.pyx

```diff
@@ -53,23 +59,33 @@ cdef extern from "precice/SolverInterface.hpp"  namespace "precice":
 
       bool isTimestepComplete()
 
+      bool hasToEvaluateSurrogateModel ()
+
+      bool hasToEvaluateFineModel ()
+
+      # action methods
+
       bool isActionRequired (const string& action)
 
       void fulfilledAction (const string& action)
 
+      # mesh access
+
       bool hasMesh (const string& meshName ) const
 
       int getMeshID (const string& meshName)
 
       set[int] getMeshIDs ()
 
-      bool hasData (const string& dataName, int meshID) const
+      # MeshHandle getMeshHandle (const string& meshName)
```

Comment:
Implementation of `MeshHandle` is not trivial, therefore we will leave it as a stub.